contracts/libraries/CommonLib.sol

```diff
@@ -75,4 +76,8 @@ library CommonLib {
         result := mload(add(source, 32))
     }
   }
+
+  function isNotGivenNFTS(uint64 maxCount, uint64 factCount) internal pure returns (bool) {   /// name!!!!
```

Comment:
`isNotGivenNFTS` -> `isAchievementAvailable`. I prefer we stick with the name `achievement` to not confuse things.